config/config_test.go

```diff
@@ -16,11 +16,23 @@ package config
 
 import (
 	"errors"
+	"fmt"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 )
 
+type NopConfig struct {
```

Comment:
No need to be public.

Reply:
I'll move 

> Let's merge as first draft, I will followup with some cleanups.

@bogdandrutu , I am going to expand the same checks to all the other type of components in the next PR. Sounds good to you?